sphinx/ext/extlinks.py

```diff
@@ -34,13 +35,54 @@
 import sphinx
 from sphinx.application import Sphinx
 from sphinx.locale import __
+from sphinx.transforms.post_transforms import SphinxPostTransform
 from sphinx.util import logging
 from sphinx.util.nodes import split_explicit_title
 from sphinx.util.typing import RoleFunction
 
 logger = logging.getLogger(__name__)
 
 
+class ExternalLinksChecker(SphinxPostTransform):
+    """
+    For each external link, check if it can be replaced by an extlink.
+
+    We treat each ``reference`` node without ``internal`` attribute as an external link.
+    """
+
+    default_priority = 900
```

Comment:
IMO, there is no special reason to invoke this feature earlier or later. So it would be better to put it on the normal priority: 500. I'll modify it after merging.